controllers/storageclient_controller.go

```diff
@@ -203,9 +205,54 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 	}
 
+	if err := r.reconcileClusterResourceQuota(externalClusterClient); err != nil {
+		return reconcile.Result{}, err
+	}
+
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClientReconciler) reconcileClusterResourceQuota(externalClusterClient *providerClient.OCSProviderClient) error {
+
+	scResponse, err := externalClusterClient.GetStorageConfig(r.ctx, r.storageClient.Status.ConsumerID)
+	if err != nil {
+		return fmt.Errorf("failed to get StorageConfig for setting quota: %v", err)
+	}
+	for _, eResource := range scResponse.ExternalResource {
+		if eResource.Kind == "ClusterResourceQuota" {
+
+			data := map[string]string{}
+			err := json.Unmarshal(eResource.Data, &data)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal data: %v", err)
+			}
+			clusterResourceQuotaResource, ok := data["clusterResourceQuota"]
+			if !ok {
+				return fmt.Errorf("no key found in the map")
+			}
+			var clusterResourceQuota quotav1.ClusterResourceQuota
+			err = json.Unmarshal([]byte(clusterResourceQuotaResource), &clusterResourceQuota)
+			if err != nil {
+				return fmt.Errorf("error: %v", err)
+			}
```

Comment:
All of this will have to change after addressing the comments here: 
https://github.com/red-hat-storage/ocs-operator/pull/2685#pullrequestreview-2176607686